File: examples/v2/service/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
service_suffix = config.get("serviceSuffix") or "dev"

ps_v2.Service(
    "catalogService",
    org_name=service_org,
    name=f"v2-service-{service_suffix}",
    description="An example v2 service catalog entry.",
    owner_type="team",
    owner_name="platform",
    items=[
        {"kind": "stack", "ref": "service-provider-test-org/example-app/dev"},
    ],
    properties=[
        {"key": "tier", "value": "gold"},
        {"key": "oncall", "value": "platform-ops"},
    ],
)

File: examples/v2/service/go/main.go
package main

import (
	v2 "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		serviceSuffix := cfg.Get("serviceSuffix")
		if serviceSuffix == "" {
			serviceSuffix = "dev"
		}

		_, err := v2.NewService(ctx, "catalogService", &v2.ServiceArgs{
			OrgName:     pulumi.String(serviceOrg),
			Name:        pulumi.String("v2-service-" + serviceSuffix),
			Description: pulumi.String("An example v2 service catalog entry."),
			OwnerType:   pulumi.String("team"),
			OwnerName:   pulumi.String("platform"),
			Items: pulumi.Array{
				pulumi.Map{
					"kind": pulumi.String("stack"),
					"ref":  pulumi.String("service-provider-test-org/example-app/dev"),
				},
			},
			Properties: pulumi.Array{
				pulumi.Map{"key": pulumi.String("tier"), "value": pulumi.String("gold")},
				pulumi.Map{"key": pulumi.String("oncall"), "value": pulumi.String("platform-ops")},
			},
		})
		return err
	})
}
